starter-code/src/viking.js

```diff
@@ -1,11 +1,109 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health,strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength;
+    }
+    receiveDamage(damage) {
+        this.health-=damage;
+
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor (name,health,strength) {
+        super(health,strength);
+        this.name = name;
+    }
+    receiveDamage(damage) {
+        this.health-=damage;
+        if(this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        } else {
+            return `${this.name} has died in act of combat`;
+        }
+    }
+    battleCry () {
+        return 'Odin Owns You All!'
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health-=damage;
+        if(this.health > 0) {
```

Comment:
Also can be return this.health > 0 ? `A Saxon has received ${damage} points of damage` :  `A Saxon has died in combat`